dtfs-central-api/src/v1/controllers/utilisation-report-service/post-report-data-validation.controller/index.test.ts

```diff
@@ -0,0 +1,121 @@
+  constructor(status?: number, message?: string) {
+    super({ status: status ?? HttpStatusCode.InternalServerError, message: message ?? '' });
+  }
+}
+
+describe('post-report-data-validation.controller', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('postReportDataValidation', () => {
+    const aValidRequestBody = (): PostReportDataValidationPayload => ({
+      reportData: [{ header: { value: 'value', column: 'A', row: '1' } }],
+    });
+
+    it('return 200 (Ok) and validation errors when successfully validates the report data', () => {
+      // Arrange
+      const reportData = [{ header: { value: 'value', column: 'A', row: '1' } }];
+      const req = httpMocks.createRequest<PostReportDataValidationRequest>({
+        body: { reportData },
+      });
+      const res = httpMocks.createResponse();
+
+      jest.mocked(validateUtilisationReportCsvData).mockReturnValue([{ errorMessage: 'Data invalid!' }]);
+
+      // Act
+      postReportDataValidation(req, res);
+
+      // Assert
+      expect(validateUtilisationReportCsvData).toHaveBeenCalledWith(reportData);
+      expect(res._getStatusCode()).toBe(HttpStatusCode.Ok);
+      expect(res._getData()).toEqual({ csvValidationErrors: [{ errorMessage: 'Data invalid!' }] });
+    });
+
+    it("responds with the specific error status if saving the report throws an 'ApiError'", () => {
+      // Arrange
+      const req = httpMocks.createRequest<PostReportDataValidationRequest>({
+        body: aValidRequestBody(),
+      });
+      const res = httpMocks.createResponse();
+
+      const errorStatus = HttpStatusCode.BadRequest;
+      jest.mocked(validateUtilisationReportCsvData).mockImplementation(() => {
+        throw new TestApiError(errorStatus, undefined);
+      });
+
+      // Act
+      postReportDataValidation(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toBe(errorStatus);
+    });
+
+    it("responds with the specific error message if saving the report throws an 'ApiError'", () => {
+      // Arrange
+      const req = httpMocks.createRequest<PostReportDataValidationRequest>({
+        body: aValidRequestBody(),
+      });
+      const res = httpMocks.createResponse();
+
+      const errorMessage = 'Some error message';
+      jest.mocked(validateUtilisationReportCsvData).mockImplementation(() => {
+        throw new TestApiError(undefined, errorMessage);
+      });
+
+      // Act
+      postReportDataValidation(req, res);
+
+      // Assert
+      expect(res._getData()).toBe(`Failed to validate report data: ${errorMessage}`);
+    });
+
+    it("responds with a '500' if an unknown error occurs", () => {
+      // Arrange
+      const req = httpMocks.createRequest<PostReportDataValidationRequest>({
+        body: aValidRequestBody(),
+      });
+      const res = httpMocks.createResponse();
+
+      jest.mocked(validateUtilisationReportCsvData).mockImplementation(() => {
+        throw new Error('Some error');
```

Comment:
This is only repeated once the type of error/message are crucial to the assertions in the tests in these files so I think we lose a fair amount of readability and maintainability for not much gain in value.

Reply:
I agree a DRY fixture for all tests for such errors would be nice, e.g `mockErrorMessage` imported from a mock/fixtures file, to avoid having the same "Some error" string repeated everywhere

But this is a tech improvement that's not part of this ticket. If we're going to create a DRY error message, we should do it it one swoop.